app/src/main/java/com/example/tasky/feature_login/presentation/LoginViewModel.kt

```diff
@@ -1,12 +1,51 @@
 package com.example.tasky.feature_login.presentation
+
+import android.util.Log
 import androidx.lifecycle.ViewModel
-import com.example.tasky.common.domain.isValidEmail
+import androidx.lifecycle.viewModelScope
+import com.example.tasky.common.data.EmailPatternValidatorImpl
+import com.example.tasky.common.data.util.Result
 import com.example.tasky.common.domain.isValidName
 import com.example.tasky.common.domain.isValidPassword
+import com.example.tasky.feature_login.data.model.LoginUserInfo
+import com.example.tasky.feature_login.data.model.RegisterUserInfo
```

Comment:
Here, you're coupling your data layer to presentation which is not allowed with this type of architecture. The direction of dependencies is this:
Presentation -> Domain <- Data

There's no arrow from presentation to data, so you should also have no imports of data in presentation (or vice versa). Instead of accessing this login and register info here, you can just pass the required strings to the data layer in this case. If it's a more complex class with many fields, you would create a domain model for this which you are allowed to access here.